tests/upgrade/0.8.2-0.9.0/jiva/suite_test.go

```diff
@@ -0,0 +1,234 @@
+		pvcName,
+		nsName,
+	)
+
+	By("verifying controller pod count ")
+	controllerPodCount := ops.GetPodRunningCountEventually(nsName, ctrlLabel, replicaCount)
+	Expect(controllerPodCount).To(Equal(replicaCount), "while checking controller pod count")
+
+	By("verifying replica pod count ")
+	replicaPodCount := ops.GetPodRunningCountEventually(nsName, replicaLabel, replicaCount)
+	Expect(replicaPodCount).To(Equal(replicaCount), "while checking replica pod count")
+
+	By("verifying status as bound")
+	status := ops.IsPVCBound(pvcName)
+	Expect(status).To(Equal(true), "while checking status equal to bound")
+
+})
+
+var _ = AfterSuite(func() {
+
+	By("deleting above pvc")
+	err := ops.PVCClient.Delete(pvcName, &metav1.DeleteOptions{})
+	Expect(err).To(
+		BeNil(),
+		"while deleting pvc {%s} in namespace {%s}",
+		pvcName,
+		nsName,
+	)
+
+	By("verifying controller pod count")
+	controllerPodCount := ops.GetPodRunningCountEventually(nsName, ctrlLabel, 0)
+	Expect(controllerPodCount).To(Equal(0), "while checking controller pod count")
+
+	By("verifying replica pod count")
+	replicaPodCount := ops.GetPodRunningCountEventually(nsName, replicaLabel, 0)
+	Expect(replicaPodCount).To(Equal(0), "while checking replica pod count")
+
+	By("deleting storageclass")
+	err = ops.SCClient.Delete(scName, &metav1.DeleteOptions{})
+	Expect(err).To(BeNil(), "while deleting storageclass {%s}", scObj.Name)
+
+	By("cleanup")
+	deleteYAML(jobYAML, "job")
+	deleteYAML(runtaskYAML, "")
+	deleteYAML(crYAML, "")
+	deleteYAML(rbacYAML, "")
+	deleteYAML(openebsYAML, "")
+	podList, err := ops.PodClient.
+		WithNamespace("default").
+		List(metav1.ListOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+	for _, po := range podList.Items {
+		if po.Status.Phase == "Succeeded" {
+			err = ops.PodClient.Delete(po.Name, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting completed pods")
+		}
+	}
+})
+
+func getYAMLFromURL(url string) string {
+	// Read yaml file from the url
+	o, err := http.Fetch(url)
+	Expect(err).ShouldNot(HaveOccurred())
+	defer o.Close()
+	b, err := ioutil.ReadAll(o)
+	Expect(err).ShouldNot(HaveOccurred())
+	// Connvert the yaml to unstructured objects
+	return string(b)
+}
+
+func applyYAML(yamlString string, flag string) {
+	unstructList, err := unstruct.ListBuilderForYamls(yamlString).Build()
+	Expect(err).ShouldNot(HaveOccurred())
+	// Applying unstructured objects
+	for i, us := range unstructList {
+		if flag == "job" && i == 0 {
+			unstructured.SetNestedStringMap(us.Object.Object, data, "data")
+		}
+		if flag == "job" && i == 1 {
+			us.Object.SetNamespace("default")
+		}
+		err = ops.UnstructClient.Create(us.Object)
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+}
+
+func deleteYAML(yamlString string, flag string) {
+	unstructList, err := unstruct.ListBuilderForYamls(yamlString).Build()
+	Expect(err).ShouldNot(HaveOccurred())
+	// Applying unstructured objects
+	for i, us := range unstructList {
+		if flag == "job" && i == 1 {
```

Comment:
This is code smell.
We need to avoid this.

Reply:
This is because the original yaml is missing the namespace for the job and unstruct will give error if not present.

Reply:
We can avoid this by just adding the namespace to the job in the yaml. I have raised a PR for that.

Reply:
Let the upgrade job be without any namespace. We should not use default namespace as some installations may not have it

Reply:
If we do not provide namespace then unstructured will give an error.